### ISSUE TITLE
Question about filter and string column's usage

### ISSUE THREAD

**Issue by HansBug** [josiahcarlson/rom#137]
The version of python: `3.5.2`
The version of redis(both client and server): `3.0.6`
The version of rom: `0.42.6`

As the following code:

```python
from rom import *

# util.set_connection_settings(**REDIS_CONFIG)

class Cell(Model):
     name = String(required=True, unique=True, default=b"", index=True, keygen=SIMPLE_CI)

_cell = Cell(name="xxx")
_cell.save()

print(_cell.name)   # it prints b'xxx'
print(Cell.query.all())   # it prints [<__main__.Cell object at 0x7f72fc5a8278>]
print(Cell.query.all()[0].name)   # it prints b'xxx'
print(Cell.query.filter(name="xxx").all())   # is prints [], why?
print(Cell.query.filter(name=b'xxx').all())   # it prints [] as well
```

May I ask how is filter used? I just followed the documentation, but it does not work.(And I'm sure that the redis server is connected properly)

And also I found that String column value is not allowed to use str format but only bytes format. Is there any better usage? or solution?

```python
_cell.name = "xxx"   # it will raise rom.exceptions.InvalidColumnValue
_cell.name = b'xxx'   # work properly, but name is String, emmmmm.....
```

**Comment by HansBug** [comment]
I know that we can use get_by, like this

```python
Cell.get_by(name="xxx")   # raise AttributeError
Cell.get_by(name="xxx".decode())   # work properly
```
But sometimes we need to get all the results, and maybe get count()'s result.

Also, like the problem above, as the README said, 

```python
user = User.get_by(email='user@host.com')
at_gmail = User.query.endswith(email='@gmail.com').all()
```

but actually `Cell.get_by(name="xxx")` can not run properly.

**Comment by josiahcarlson** [comment]
Sorry, just getting to this.

> _cell.name = b'xxx'   # work properly, but name is String, emmmmm.....

1. See the docs for: https://josiahcarlson.github.io/rom/columns.html#rom.columns.String
2. See the docs for: https://josiahcarlson.github.io/rom/columns.html#rom.columns.Text

> May I ask how is filter used? I just followed the documentation, but it does not work.(And I'm sure that the redis server is connected properly)

1. See the docs for https://josiahcarlson.github.io/rom/util.html#rom.util.SIMPLE_CI
2. See the docs for https://josiahcarlson.github.io/rom/util.html#rom.util.SIMPLE

This doesn't work the way you expect, because this keygen is for *ordering* not *searching*. If you want to search by the value of a string:

1. https://josiahcarlson.github.io/rom/util.html#rom.util.IDENTITY
2. https://josiahcarlson.github.io/rom/util.html#rom.util.IDENTITY_CI

Now in this case, unique already exists, so is a type of index that can be used by `Model.get_by()`, so when I run this:
```
from rom import *

# util.set_connection_settings(**REDIS_CONFIG)

class Cell(Model):
     name = String(required=True, unique=True, default=b"", index=True, keygen=SIMPLE_CI)

for i in Cell.query.all():
    i.delete()

session.commit()

_cell = Cell(name="xxx")
_cell.save()

print(_cell.name)   # it prints b'xxx'
print(Cell.query.all())   # it prints [<__main__.Cell object at 0x7f72fc5a8278>]
print(Cell.query.all()[0].name)   # it prints b'xxx'
print(Cell.query.filter(name="xxx").all())   # is prints [], why?
print(Cell.query.filter(name=b'xxx').all())   # it prints [] as well

print(Cell.get_by(name=b'xxx'))   # prints <__main__.Cell object at 0x7f12b0c593c8>
```
I get:
```
$ python3.6 test_137.py
b'xxx'
[<__main__.Cell object at 0x7f6dc78b8438>]
b'xxx'
[]
[]
<__main__.Cell object at 0x7f6dc78b8438>
```

But when I run this:
```
class Cell2(Model):
    name = String(required=True, unique=True, default=b"", index=True, keygen=IDENTITY_CI)

for i in Cell2.query.all():
    i.delete()

_cell = Cell2(name="xxx")
_cell.save()

print(_cell.name)   # it prints b'xxx'
print(Cell2.query.all())   # it prints [<__main__.Cell object at 0x7f72fc5a8278>]
print(Cell2.query.all()[0].name)   # it prints b'xxx'
print(Cell2.query.filter(name="xxx").all())   # is prints [<__main__.Cell2 object at 0x7f6dc7242668>]
print(Cell2.query.filter(name=b'xxx').all())   # it prints [<__main__.Cell2 object at 0x7f6dc7242668>]
print(Cell2.get_by(name=b'xxx'))   # prints <__main__.Cell object at 0x7f12b0c593c8> (can use 2 indexes)
```
I get:
```
b'xxx'
[<__main__.Cell2 object at 0x7fc25857e668>]
b'xxx'
[<__main__.Cell2 object at 0x7fc25857e668>]
[<__main__.Cell2 object at 0x7fc25857e668>]
<__main__.Cell2 object at 0x7fc25857e668>
```

Long story short: use the right index keygen, get the results you expect. Use the wrong index keygen, get things you don't expect.

**Comment by josiahcarlson** [comment]
Oh, and for:

> `print(_cell.name)   # it prints b'xxx`

... we do some magic on instantiation to sometimes allow unicode strings as arguments to byte string arguments, out of convenience. But if you're not in-line creating an object (setting an attribute, or loading from Redis itself), you may need to be more careful.

**Comment by josiahcarlson** [comment]
> but actually `Cell.get_by(name="xxx")` can not run properly.

Please read the docs I linked. And if you still feel ^^^, please read:
https://github.com/josiahcarlson/rom/blob/master/rom/model.py#L576 (in current master) to see how the magic works, even with your original code.
